### ISSUE TITLE
[4.0.0] indent multiline expression interacting poorly with ArrayExpression/ObjectExpression

### ISSUE THREAD

**Issue by Pyrolistical** [eslint/eslint#8717]
**Tell us about your environment**

* **ESLint Version:** 4.0.0
* **Node Version:** 6.9.1
* **npm Version:** 4.6.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
https://github.com/Pyrolistical/eslint-4-indent-inline-members

```
{
  "rules": {
    "indent": [
      "error",
      2
    ]
  },
  "env": {
    "es6": true
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
a([
  b
]) || [
  c
]

a({
  b
}) || {
  c
}
```

**What did you expect to happen?**
No errors

**What actually happened? Please include the actual, raw output from ESLint.**
```
   4:1  error  Expected indentation of 4 spaces but found 2  indent
   5:1  error  Expected indentation of 2 spaces but found 0  indent
  10:1  error  Expected indentation of 4 spaces but found 2  indent
  11:1  error  Expected indentation of 2 spaces but found 0  indent

✖ 4 problems (4 errors, 0 warnings)
  4 errors, 0 warnings potentially fixable with the `--fix` option.
```

This is related to the indent rewrite. This worked fine in eslint 3.x.

I want MemberExpression to remain at 1 to enforce this:
```
foo
  .bar
  .baz();
```

I believe this is a bug, because if you set ArrayExpression to zero, then line 2 will error.

**Comment by Pyrolistical** [comment]
also, I want MemberExpression to remain at 1 to enforce this:
```
foo
  .bar
  .baz();
```

**Comment by not-an-aardvark** [comment]
Thanks for the report. I agree that this seems like a bug.

As a sidenote, I don't think this is related to the MemberExpression option, because your example doesn't contain any MemberExpressions. (A MemberExpression is a property access like `foo.bar`.)

**Comment by Pyrolistical** [comment]
You're right. I thought changing MemberExpression would change the result, but it didn't.
